synapse/util/logcontext.py

```diff
@@ -116,8 +116,6 @@ def set_current_context(cls, context):
 
     def __enter__(self):
         """Enters this logging context into thread local storage"""
-        if self.parent_context is not None:
-            raise Exception("Attempt to enter logging context multiple times")
         self.parent_context = self.set_current_context(self)
```

Comment:
Maybe we shouldn't call this the ``parent_context`` anymore. Something like ``previous_context`` might be more appropriate?